### PR TITLE
Add the --default_data_plane_device netdev command line argument

### PR DESCRIPTION
**Pull request by oserres**

### CODE REVIEW

File: distbench_netutils.cc [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    if (family == AF_PACKET)
+      continue;
+
+    s = getnameinfo(ifa->ifa_addr,
+                    (family == AF_INET) ? sizeof(struct sockaddr_in) :
+                                          sizeof(struct sockaddr_in6),
+                    host, NI_MAXHOST, NULL, 0, NI_NUMERICHOST);
+    if (s == 0) { // Success
+      DeviceIpAddress ip_address(host, ifa->ifa_name, family);
+      result.push_back(ip_address);
```
Comment (danmanj) [inline distbench_netutils.cc:63]: emplace_back

File: distbench_netutils.cc [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+               << " prefer_ipv4=" << prefer_ipv4;
+  }
+
+  int score = 0;
+  DeviceIpAddress best_match;
+  for (const auto& address: all_addresses) {
+    int cur_score = 0;
+    if (address.isIPv4() == prefer_ipv4)
+      cur_score += 2048;
+    if (address.netdevice() == netdev)
```
Comment (danmanj) [inline distbench_netutils.cc:92]: this can never evaluate to true

File: distbench_netutils.cc [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  auto all_addresses = GetAllAddresses();
+
+  // Exact match
+  for (const auto& address: all_addresses) {
+    if (address.isIPv4() == prefer_ipv4 &&
+        address.netdevice() == netdev)
+      return address;
+  }
+
+  if (!netdev.empty()) {
```
Comment (danmanj) [inline distbench_netutils.cc:81]: before giving up run a second pass examining only the netdev, and ignoring v4/v6.

**Review by danmanj — APPROVED** [review]
Please make a few tweaks.

File: distbench_netutils.cc (hunk 1 above)
Comment (oserres) [reply distbench_netutils.cc]: Done

File: distbench_netutils.cc (hunk 2 above)
Comment (oserres) [reply distbench_netutils.cc]: Removed

File: distbench_netutils.cc (hunk 3 above)
Comment (oserres) [reply distbench_netutils.cc]: Done; displaying a warning

**Pull request merged**